### PR TITLE
test(eighth): add rooms, attendance and signup tests

### PR DESCRIPTION
**Pull request by Laur04**

### CODE REVIEW

File: intranet/apps/eighth/tests.py [hunk 1]
```diff
@@ -428,6 +456,34 @@ def test_take_attendance_google_meet_csv(self):
...
 
+        # Make sure bad file fails nicely with KeyError
+        with tempfile.NamedTemporaryFile(mode="w+") as f:
+            writer = csv.DictWriter(f, fieldnames=["NotName", "NotEmail"])
+            writer.writeheader()
+            writer.writerow({"NotName": "Test User", "NotEmail": "12345@fcpsschools.net"})
+            f.seek(0)
+            response = self.client.post(reverse("eighth_take_attendance", args=[schact1.id]), {"attendance": f}, follow=True)
+
+        self.assertEqual(response.status_code, 200)
```
Comment (etnguyen03) [inline intranet/apps/eighth/tests.py:467]: I think we should check for the presence of the "could not interpret file" message.

```suggestion
            self.assertIn("Could not interpret file. Did you upload a Google Meet attendance report without modification?", list(map(str, list(response.context["messages"]))))
            self.assertEqual(response.status_code, 200)
```

File: intranet/apps/eighth/tests.py [hunk 2]
```diff
@@ -428,6 +456,34 @@ def test_take_attendance_google_meet_csv(self):
...
+
+        # Make sure bad file fails nicely with IndexError
+        with tempfile.NamedTemporaryFile(mode="w+") as f:
+            writer = csv.DictWriter(f, fieldnames=["Name", "Email"])
+            writer.writeheader()
+            writer.writerow({"Name": "User", "Email": "@fcpsschools.net"})
+            f.seek(0)
+            response = self.client.post(reverse("eighth_take_attendance", args=[schact1.id]), {"attendance": f}, follow=True)
+
```
Comment (etnguyen03) [inline intranet/apps/eighth/tests.py:476]: I think we should check for the presence of the "could not interpret file" message, and probably check for a 200 code here too

```suggestion
            self.assertIn("Could not interpret file. Did you upload a Google Meet attendance report without modification?", list(map(str, list(response.context["messages"]))))
            self.assertEqual(response.status_code, 200)
```

File: intranet/apps/eighth/tests.py [hunk 3]
```diff
@@ -428,6 +456,34 @@ def test_take_attendance_google_meet_csv(self):
...
+        # Make sure bad file fails nicely with ValueError
+        with tempfile.NamedTemporaryFile(mode="w+") as f:
+            writer = csv.DictWriter(f, fieldnames=["Name", "Email"])
+            writer.writeheader()
+            writer.writerow({"Name": 1, "Email": 5})
+            f.seek(0)
+            response = self.client.post(reverse("eighth_take_attendance", args=[schact1.id]), {"attendance": f}, follow=True)
+
+        self.assertEqual(response.status_code, 200)
```
Comment (etnguyen03) [inline intranet/apps/eighth/tests.py:485]: Same here too

```suggestion
            self.assertIn("Could not interpret file. Did you upload a Google Meet attendance report without modification?", list(map(str, list(response.context["messages"]))))
            self.assertEqual(response.status_code, 200)
```

**Review by etnguyen03 — CHANGES_REQUESTED** [review]

**Comment by etnguyen03** [comment]
Hmm... maybe we should break up the `tests` module I suppose?

**Comment by Laur04** [comment]
Yea I guess we need to. According to [this](https://stackoverflow.com/questions/6248510/how-to-spread-django-unit-tests-over-multiple-files), it isn't that complicated.

**Review by etnguyen03 — APPROVED** [review]
LGTM

**Pull request merged**